File: schemaorg/rsvpresponsetype.go
package schemaorg

import "encoding/json"

// RsvpResponseType see : https://schema.org/RsvpResponseType
type RsvpResponseType struct {

typeContext

Enumeration

// SupersededBy see : https://schema.orghttp://meta.schema.org/supersededBy
// Relates a term (i.e. a property, class or enumeration) to one that supersedes it.
SupersededBy interface{} `json:"supersededBy"` // types : Class Enumeration Property

}

func (v *RsvpResponseType) MarshalJSON() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "RsvpResponseType"

	return json.Marshal(v)
}

File: schemaorgphp/classes/rsvp_response_type.php
<?php

class RsvpResponseType extends Enumeration implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'RsvpResponseType';
	
	/**
	 * Relates a term (i.e. a property, class or enumeration) to one that supersedes it.
	 * see : https://meta.schema.org/supersededBy
	 * @var \Class|\Class[]|\Enumeration|\Enumeration[]|\Property|\Property[]
	 */
	public var $superseded_by;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'RsvpResponseType'
		);
		
		$serialized = so_json_serialize( $this->superseded_by );
		if ( ! empty( $serialized ) ) {
			$out['supersededBy'] = $serialized;
		}
		
		return $out;
	}
}
